METHOD_NAMING_PROPOSAL.md

```diff
@@ -0,0 +1,412 @@
+googlePayClient.requestPayment(activity, request)
+```
+
+</td>
+
+<td>
+
+```kotlin
+googlePayLauncher = GooglePayLauncher(activity) { googlePayResult ->
+  // Parse Google Pay result and return a nonce
+  googlePayClient.tokenize(googlePayResult) 
+}
+
+// Create PaymentDataRequest for Google Pay SDK
+googlePayClient.requestPayment(context, request) { googlePayIntentData, error ->
+  // Launch Google Pay payment sheet
+  googlePayLauncher.launch(googlePayIntentData) 
+}
+```
+
+</td>
+
+</tr>
+</table>
+
+
+### Consistent Naming Pattern
+
+With this implementation, the method names do not align with v4, so there is greater merchant migration difficulty, but the naming pattern is consistent across payment methods and provides clarity to what is actually occurring.
+For this pattern, I've chosen to use the `paymentAuth` naming pattern.
+
+`client.createPaymentAuthRequest` returns `paymentAuthRequest`
+
+`launcher.launch(paymentAuthRequest)` returns `paymentAuthResult`
+
+`client.tokenize(paymentAuthResult)` returns `paymentMethodNonce`
+
+This could also be called `paymentAuthorization`, `authChallenge`, `authenticationChallenge`, `paymentFlow`, `paymentSheet`, or any other suggested pattern. This naming should be able to be applied to launching an external browser, launching an external app, presenting a payment sheet. The user may or may not need to interact with these external flows to complete the payment flow (ex: Venmo app launches and returns without interaction, 3DS might require a code to be input).
+
+<table>
+<tr>
+<th>v4 PayPal</th>
+<th>v5 PayPal</th>
+</tr>
+<tr>
+
+<td>
+
+```kotlin 
+// Retrieve data required to launch browser
+// Launch browser flow, parse browser result
+// Return nonce to listener
+payPalClient.tokenizePayPalAccount(activity, request)
+```
+
+</td>
+
+<td>
+
+```kotlin
+payPalLauncher = PayPalLauncher() { payPalPaymentAuthResult ->
+  // Parse the result from the PayPal browser flow and return a nonce
+  payPalClient.tokenize(payPalPaymentAuthResult) 
+}
+
+// Retrieve data required to launch the PayPal flow in a browser
+payPalClient.createPaymentAuthRequest(context, request) { payPalPaymentAuthRequest, error ->
+  // Launch the PayPal in a browser
+  payPalLauncher.launch(activity, payPalPaymentAuthRequest) 
+}
+```
+
+</td>
+
+</tr>
+
+<tr>
+<th>v4 Local Payment</th>
+<th>v5 Local Payment</th>
+</tr>
+<tr>
+
+<td>
+
+```kotlin 
+// Retrieve data required to start local payment authorization
+// Or return result if no additional auth required
+localPaymentClient.startPayment(activity, request) { localPaymentResult, error ->
+  // Initiate browser flow and return nonce to listener
+  localPaymentClient.approveLocalPayment(activity, localPaymentResult)
+}
+```
+
+</td>
+
+<td>
+
+```kotlin
+localPaymentLauncher = LocalPaymentLauncher() { localPaymentAuthResult ->
+  // Parse the result from the local payment browser flow and return a nonce
+  localPaymentClient.tokenize(localPaymentAuthResult) 
```

Comment:
I like how this proposal keeps `tokenize()` as the last step in all of the flows!